### PR TITLE
added TOTP call

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: src/ipc/totp-get.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+beforeEach(() => {
+  execMock.mockReset()
+  execMock.mockResolvedValue({ stdout: '', stderr: '' })
+})
+
+const { ipcMain, ipcRenderer } = fakeIpc()
+register(ipcMain)
+
+test('call to totp calls appropriate shell command and returns the right data', async () => {
```
Comment (umbernhard) [inline src/ipc/totp-get.test.ts:17]: Are you also gonna mock the case where tpm2-totp hasn't been initialized yet?

**Review by umbernhard — CHANGES_REQUESTED** [review]

File: src/ipc/totp-get.test.ts (hunk 1 above)
Comment (benadida) [reply src/ipc/totp-get.test.ts:17]: yeah, happy to, can you tell me what happens then? :)

Comment (umbernhard) [reply src/ipc/totp-get.test.ts:17]: it'll throw an error, so you'll get input on stderr but not on stdout (I think).

File: src/ipc/totp-get.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ])
+
+    const [timestamp, code] = stdout.split(': ')
+
+    return {
+      isoDatetime: new Date(timestamp).toISOString(),
+      code,
+    }
+  } catch {
+    return undefined
```
Comment (eventualbuddha) [inline src/ipc/totp-get.ts:26]: Also log error with `debug`?

File: src/ipc/totp-get.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+async function totpGet(): Promise<TotpInfo | undefined> {
+  try {
+    const { stdout, stderr } = await exec('sudo', [
+      '-n',
+      '/usr/local/bin/tpm2-totp',
+      'show',
+    ])
+
+    const [timestamp, code] = stdout.split(': ')
```
Comment (eventualbuddha) [inline src/ipc/totp-get.ts:19]: How is this working? Won't this split `HH:MM:SS`?

Comment (eventualbuddha) [reply src/ipc/totp-get.ts:19]: Oh, I see, the space. 🤔

File: src/ipc/totp-get.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+export interface TotpInfo {
+  isoDatetime: string
+  code: string
+}
+
+async function totpGet(): Promise<TotpInfo | undefined> {
+  try {
+    const { stdout, stderr } = await exec('sudo', [
+      '-n',
+      '/usr/local/bin/tpm2-totp',
```
Comment (umbernhard) [inline src/ipc/totp-get.ts:18]: I think we also need the `-` flag here to get the timestamp.

**Review by umbernhard — CHANGES_REQUESTED** [review]

File: src/ipc/totp-get.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    if (stderr) return undefined
+
+    const [timestamp, code] = stdout.split(': ')
+
+    return {
+      isoDatetime: new Date(timestamp).toISOString(),
+      code,
+    }
+  } catch (err) {
+    debug('could not call tpm2-totp', err)
```
Comment (eventualbuddha) [inline src/ipc/totp-get.ts:31]: `tpm2-totp: %s', err)`

Comment (benadida) [reply src/ipc/totp-get.ts:31]: this worked fine fwiw, but happy to change it :)

**Review by umbernhard — APPROVED** [review]

**Pull request merged**